### ISSUE TITLE
Статус ЭД

### ISSUE THREAD

**Issue by nikitaryanov** [1C-Company/DirectBank#46]
Добрый день!

Подскажите, пожалуйста, по поводу статуса ЭД, его изменения в ходе обработки и получения. Интересуют следующие моменты:
1. Верно ли я понимаю, что когда мы получили любой ЭД на обработку, то должны изменить его статус на "Принят", а затем, когда закончили его обработку, на "Исполнен"?
2. В описании процесса получения статуса документа написано:

> Если запрос корректный, то на стороне Банка выполняется запрос на получение актуального статуса исходного электронного документа, затем формируется электронный документ «Извещение о состоянии электронного документа» (XML-файл, соответствующий XML-схеме извещения о состоянии электронного документа), содержащий 2 идентификатора (ИД исходного электронного документа и ИД запроса), а также, либо ошибку обработки исходного электронного документа, либо его текущий статус.

Что подразумевается под `либо ошибку обработки исходного электронного документа, либо его текущий статус`? Допустим, при обработке исходного ЭД возникла ошибка проверки ЭЦП и мы подготовили ответ с описанием ошибки и изменили статус на 03 (отклонен банком). Что должен получить клиент в ответ на запрос статуса ЭД? Код 03 или код, связанный с ошибкой проверки ЭЦП, учитывая, что код с описанием ошибки клиент получит в любом случае при попытке получить документ через GetPack? 
3. Исходя из предыдущего вопроса, в каких ситуациях может понадобиться статус 03?

**Comment by SevDmitry** [comment]
> Верно ли я понимаю, что когда мы получили любой ЭД на обработку, то должны изменить его статус на "Принят", а затем, когда закончили его обработку, на "Исполнен"?

Верно

Если есть проблемы со чтением электронного документа или проверкой подписи - то это ошибка первичной обработки документа, и желательно возвращать извещение с элементом Error. Если первичная проверка документа выполнена и он принят к исполнению, то нужно возвращать извещение с элементом Status. Статус 03 может понадобиться, например, если получателя платежа не существует, т.к. клиент ошибся в реквизитах. Описание статуса поменяю, так как это не первичный статус.

**Comment by nikitaryanov** [comment]
@SevDmitry спасибо!

Тогда еще один вопрос:
- Получается, что в случае успешной обработки ЭД клиент может получить как результат обработки (например, выписки), так и статус - Accepted/Done, и это будут два разных документа.
В случае же с ошибкой во время обработки, клиент получит два одинаковых документа (StatusDocNotice) на два разных запроса: на запрос статуса документа и на запрос результата. Верно ли мое предположение?

**Comment by SevDmitry** [comment]
> Верно ли мое предположение?
Верно. Но запрос статуса пользователь делает только вручную. Если сразу после отправки документа придет извещение с ошибкой, то пользователь даже не сможет запросить статус такого документа.

> Сообщения об ошибке или в описании статусов, которые написаны в документации, носят рекомендательный характер или их содержимое можно менять на усмотрение банка?
Рекомендательный характер. Но смысл текста должен быть как в стандарте, если совпадает код. 
Например, при получении кода 1006 мы снова запускаем процесс аутентификации. На все другие случаи возвращайте свои коды и описание.

**Comment by nikitaryanov** [comment]
Спасибо!

**Comment by BSSAnalytics** [comment]
Добрый день.
В процессе обработки в банке бывают разные ситуации. И иногда случается так, что документ (например, платежное поручение) автоматически отказывается АБС и в 1С в таком случае уходит статус 03. Далее сотрудники банка в ручном режиме возвращают документ на повторную обработку, и документ успешно исполняется.
Если я правильно понял ответы выше, то клиент уже не сможет запросить статус документа а банке (он может и не знать, что это надо сделать, если еще не обратился в банк). Подскажите, если в случае очередного сеанса обмена банковская ИС отдаст "Извещение о состоянии электронного документа" по такому РПП с новым статусом, то 1С нормально его обработает в таком случае (т.е. ранее был получен код 03, а теперь - 02)?
